superset/db_engine_specs.py

```diff
@@ -74,6 +81,103 @@ def extra_table_metadata(cls, database, table_name, schema_name):
+                                 decimal=decimal,
+                                 quotechar=quotechar,
+                                 escapechar=escapechar,
+                                 comment=comment,
+                                 encoding='utf-8',
+                                 error_bad_lines=error_bad_lines,
+                                 chunksize=chunksize,
+                                 iterator=True)
+        df = pandas.DataFrame()
+        df = pandas.concat(chunk for chunk in chunks)
+        print(df)
+        return df
+
+    @staticmethod
+    def df_to_db(df, name, con, schema, if_exists, index,
+                 index_label, chunksize):
+
+        engine = create_engine(con, echo=False)
+
+        # Use Pandas to parse dataframe to database
+        df.to_sql(name=name, con=engine, schema=schema, if_exists=if_exists,
+                  index=index, index_label=index_label, chunksize=chunksize)
+
+
+    @staticmethod
+    def upload_csv(form):
+        print(results_backend)
+        def allowed_file(filename):
+            # Only allow specific file extensions as specified in the config
+            return '.' in filename and \
+                   filename.rsplit('.', 1)[1] in config['ALLOWED_EXTENSIONS']
+
+        def upload_file(csv_file):
+            if csv_file and csv_file.filename:
+                filename = secure_filename(csv_file.filename)
+                csv_file.save(os.path.join(config['UPLOAD_FOLDER'],
+                                                 filename))
+                return filename
+
+        filename = secure_filename(form.csv_file.data.filename)
+        df = BaseEngineSpec.csv_to_df(names=form.names.data,
+                            filepath_or_buffer=filename,
+                            sep=form.sep.data,
+                            header=form.header.data,
+                            index_col=form.index_col.data,
+                            squeeze=form.squeeze.data,
+                            prefix=form.prefix.data,
+                            mangle_dupe_cols=form.mangle_dupe_cols.data,
+                            skipinitialspace=form.skipinitialspace.data,
+                            skiprows=form.skiprows.data,
+                            nrows=form.nrows.data,
+                            skip_blank_lines=form.skip_blank_lines.data,
+                            parse_dates=form.parse_dates.data,
+                            infer_datetime_format=form.infer_datetime_format.data,
+                            dayfirst=form.dayfirst.data,
+                            thousands=form.thousands.data,
+                            decimal=form.decimal.data,
+                            quotechar=form.quotechar.data,
+                            escapechar=form.escapechar.data,
+                            comment=form.comment.data,
+                            error_bad_lines=form.error_bad_lines.data,
+                            chunksize=10000)
```

Comment:
seems arbitrary, unclear where it belongs at this point though, maybe a database configuration element? Ok leaving as is for now though.